### PR TITLE
Clean BuildFiles in EventFilter

### PR DESCRIPTION
**Pull request by guitargeek**
#### PR description:

Another quick partially automatic BuildFile cleaning PR in the style of many before (for example https://github.com/cms-sw/cmssw/pull/31165).

As always, only the dependencies which are **not included in any of the sources** are removed, so these changes are orthogonal and complementary to the recent PRs which added all packages that are **actually included**.

#### PR validation:

CMSSW compiles. It was checked that all newly added dependencies are actually required by the package with `git grep`.

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by jfernan2** [comment]
+1

**Comment by mommsen** [comment]
+1

**Comment by perrotta** [comment]
+1
- Cleaning implemented as planned
- Jenkins tests pass

File: EventFilter/GEMRawToDigi/BuildFile.xml [hunk 1]
```diff
@@ -1,15 +1,3 @@
-<use name="FWCore/Framework"/>
-<use name="DataFormats/FEDRawData"/>
-<use name="FWCore/PluginManager"/>
-<use name="FWCore/ParameterSet"/>
-<use name="CondFormats/DataRecord"/>
-<use name="CondFormats/GEMObjects"/>
-<use name="FWCore/MessageLogger"/>
-<use name="Utilities/StorageFactory"/>
-<use name="FWCore/Sources"/>
-<use name="DataFormats/Common"/>
```
Comment (slava77) [inline EventFilter/GEMRawToDigi/BuildFile.xml:10]: this was added recently in 477a01203caeebd22d174ac02e90d5600c9cbd1e
it looks like the include of DataFormats/Common/interface/Handle.h is actually in EventFilter/GEMRawToDigi/plugins
The other two deps (MuonData and MuonDetId are apparently no longer needed, OK there).

I guess some other package carries the DataFormats/Common to keep things working.

File: EventFilter/Phase2TrackerRawToDigi/BuildFile.xml [hunk 2]
```diff
@@ -1,13 +1,5 @@
-<use name="CondFormats/DataRecord"/>
-<use name="CondFormats/SiStripObjects"/>
-<use name="DataFormats/Common"/>
-<use name="DataFormats/DetId"/>
 <use name="DataFormats/FEDRawData"/>
-<use name="DataFormats/Phase2TrackerDigi"/>
```
Comment (slava77) [inline EventFilter/Phase2TrackerRawToDigi/BuildFile.xml:6]: this is somewhat similar (used in the /plugins while the /plugins build file does not have this dependency explicitly.
At least in this case there was no recent PR that added this dep

**Comment by davidlange6** [comment]
this was added recently in 477a012<https://github.com/cms-sw/cmssw/commit/477a01203caeebd22d174ac02e90d5600c9cbd1e>
it looks like the include of DataFormats/Common/interface/Handle.h is actually in EventFilter/GEMRawToDigi/plugins

The src of this package has changed considerably since that commit....

**Comment by perrotta** [comment]
@slava77 , I removed the "+1" from reco above.
Please sign it as soon as you agree with the updates implemented in this PR

**Comment by slava77** [comment]
> @slava77 , I removed the "+1" from reco above.
> Please sign it as soon as you agree with the updates implemented in this PR

from @davidlange6  https://github.com/cms-sw/cmssw/pull/31242#issuecomment-682071664
at least the main point may be not relevant .

The pattern still appears to be present in this PR:  a dependence on A is declared in Package while the actual use is in Package/plugin and this PR removes the dependence, but does not move it to Package/plugin. 
I can't recall if this was already the case in other similar PRs.
I suspect it's been the case and nothing bad happened.

**Comment by guitargeek** [comment]
Should be better now! I also merged the `.cc` and `.h` files of the plugins that Slava mentioned, so it's easier to spot duplicate includes etc.

**Comment by smorovic** [comment]
+1

**Comment by davidlange6** [comment]
> The pattern still appears to be present in this PR: a dependence on A is declared in Package while the actual use is in Package/plugin and this PR removes the dependence, but does not move it to Package/plugin.
> I can't recall if this was already the case in other similar PRs.
> I suspect it's been the case and nothing bad happened.

its "ok" in the sense that things compile and link until the requirement is not picked up by some other indirect dependency....

**Comment by kpedro88** [comment]
+upgrade

**Comment by jfernan2** [comment]
+1

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/31242	44bccaf
- additional code changes since the previous reco signoff are in line with the review comments: https://github.com/cms-sw/cmssw/compare/1fde8ed684550ddd8c1a36913136f35260fc8115...44bccaf7431b41d4303b47754f2ec0e3f698c81e
- jenkins tests pass and comparisons with the baseline show no differences

**Comment by silviodonato** [comment]
merge

**Pull request merged**